node/src/reactor.rs

```diff
@@ -443,33 +470,76 @@ where
         self.event_count += 1;
     }
 
-    // Handles dumping queue contents to files in /tmp.
-    async fn dump_queues(&self) {
+    /// Gets both the allocated and total memory from sys-info + jemalloc
+    fn get_allocated_memory() -> Option<AllocatedMem> {
+        let mem_info = match sys_info::mem_info() {
+            Ok(mem_info) => mem_info,
+            Err(error) => {
+                error!(%error, "unable to get mem_info using sys-info");
+                return None;
+            }
+        };
+
+        // mem_info gives us kb
+        let total = mem_info.total * 1024;
+
+        match jemalloc_epoch::mib() {
+            Ok(mib) => {
+                // jemalloc_ctl requires you to advance the epoch to update its stats
+                if let Err(advance_error) = mib.advance() {
+                    error!(%advance_error, "unable to advance jemalloc epoch");
+                }
+            }
+            Err(error) => {
+                error!(%error, "unable to get epoch::mib from jemalloc");
+                return None;
+            }
+        }
+        let allocated = match jemalloc_allocated::mib() {
```

Comment:
Does this value represent kB rather than bytes too?  If so, we need to multiply it by 1024 too.

Either way, it'd be good to at least document the units used for each field of the `AllocatedMem` struct I guess.

Reply:
It's bytes, but I've documented that here and in the struct definition now.